### PR TITLE
Option to hide exit button, function to force close Immersive Reader

### PR DESCRIPTION
**Pull request by pstakoff**
Adds **hideExitButton** option to determine whether or not the Immersive Reader's exit button will be visible.
Adds and exports **close()** function to force close the Immersive Reader

### CODE REVIEW

File: js/samples/advanced-nodejs/views/inner-hide-exit-button.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            <h1 id='ir-title'>Geography</h1>
+            <span id='ir-text'>
+                <p>The study of Earth’s landforms is called physical geography. Landforms can be mountains and valleys. They can also be glaciers, lakes or rivers.</p>
+                <p>The physical features of a region are often rich in resources. Within a nation, mountain ranges become natural borders for settlement areas.</p>
+            </span>
+        </article>
+    </main>
+
+    <script type='text/javascript'>
+        async function handleLaunchImmersiveReader(sampleId) {
```
Comment (YanirShahak) [inline js/samples/advanced-nodejs/views/inner-hide-exit-button.html]: >async [](start = 8, length = 5)

pretty sure this isn't supported in IE

File: js/samples/advanced-nodejs/views/inner-hide-exit-button.html [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    lang: 'en'
+                }]
+            };
+
+            const options = {
+                'hideExitButton': true,
+                'onExit': () => { console.log("You can still have a callback function on exit."); }
+            }
+
+            const token = await getImmersiveReaderTokenAsync();
```
Comment (YanirShahak) [inline js/samples/advanced-nodejs/views/inner-hide-exit-button.html]: >await [](start = 26, length = 5)

same

File: js/src/options.ts [hunk 3]
```diff
@@ -9,4 +9,5 @@ export type Options = {
     onExit?: () => any;        // Executes when the Immersive Reader exits
     customDomain?: string;     // Reserved for internal use. Custom domain where the Immersive Reader webapp is hosted (default is null).
     allowFullscreen?: boolean; // The ability to toggle fullscreen (default is true).
+    hideExitButton?: boolean; // Whether or not to hide the Immersive Reader's exit button arrow (default is false). This should only be true if there is an alternative mechanism provided to exit the Immersive Reader (e.g a mobile toolbar's back arrow).
```
Comment (YanirShahak) [inline js/src/options.ts]: >// [](start = 30, length = 2)

Super duper nit: the comment characters seem to be aligned cross all options... this is missing a single space... sorry... i couldn't resist :blush:

File: js/src/launchAsync.ts [hunk 4]
```diff
@@ -154,3 +160,7 @@ export function launchAsync(token: string, subdomain: string, content: Content,
         document.head.appendChild(noscroll);
     });
 }
+
+export function close() {
+    window.postMessage('ImmersiveReader-Exit', '*');
+}
```
Comment (YanirShahak) [inline js/src/launchAsync.ts]: We need to update the documentation for this.
In addition, do we need to listen to ImmersiveReader-Exit messages from the WebApp and relay them to the hosting app when the exit button exists and is pressed by the user?

Comment (YanirShahak) [reply js/src/launchAsync.ts]: In addition, perhaps the close() method should also remove the iframe and reset the internal state (call reset())
?

---
In reply to: [327718990](https://github.com/microsoft/immersive-reader-sdk/pull/97#discussion_r327718990) [](ancestors = 327718990)

File: js/samples/advanced-nodejs/views/inner-hide-exit-button.html (hunk 1 above)
Comment (pstakoff) [reply js/samples/advanced-nodejs/views/inner-hide-exit-button.html]: We need to update this in all the samples. Will do that next

File: js/samples/advanced-nodejs/views/inner-hide-exit-button.html (hunk 2 above)
Comment (pstakoff) [reply js/samples/advanced-nodejs/views/inner-hide-exit-button.html]: see above

File: js/src/launchAsync.ts (hunk 4 above)
Comment (pstakoff) [reply js/src/launchAsync.ts]: Going to send a PR for updating documentation. 
Line 94 is listening for "ImmersiveReader-Exit". I could move all of the logic inside of that conditional into the close() function but I think that it has the same functionality.

File: js/src/options.ts (hunk 3 above)
Comment (pstakoff) [reply js/src/options.ts]: Could catch :D

File: js/src/launchAsync.ts (hunk 4 above)
Comment (pstakoff) [reply js/src/launchAsync.ts]: https://github.com/MicrosoftDocs/azure-docs-pr/pull/89590

**Pull request merged**